pkg/generate/app/cmd/resolve.go

```diff
@@ -108,7 +106,10 @@ func Resolve(r *Resolvers, c *ComponentInputs, g *GenerationInputs) (*ResolvedCo
 	for _, repo := range repositories {
 		repo.SetContextDir(g.ContextDir)
 	}
-
+	// Environment File where the parameters are stored
+	if len(g.EnvironmentFile) != 0 && !app.IsPossibleEnvironmentFile(g.EnvironmentFile) {
+		return nil, fmt.Errorf("when --env-file is specified you must provide a valid env file location")
+	}
 	if len(g.Strategy) != 0 && len(repositories) == 0 && !g.BinaryBuild {
 		return nil, fmt.Errorf("when --strategy is specified you must provide at least one source code location")
 	}
```

Comment:
While I agree with your logic, I think it still doesn't explain validation for above lines, especially the logic for checking strategy and repository location should be moved to another function, if I understood you correctly. However, I made the changes the you suggested, now the validation logic resides in newapp.go within validate() function. Also, please note that format for env file is "**--env-file=myenvfile**" not "**--env-file myenvfile**"